### ISSUE TITLE
Binding redirects are not created for Test Projects

### ISSUE THREAD

**Issue by onovotny** [Microsoft/vstest#650]
In some cases, binding redirects are not being generated to test projects when they are needed. This is because the following two properties are missing from the default inferred targets:

```
  <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
  <GenerateBindingRedirectsOutputType>true</GenerateBindingRedirectsOutputType>
```
This can come up if there are P2P references between a .NET 4.5 project that uses System.Net.Http and a .NET 4.6.1 test project that has any possible transitive ref to a newer http client lib. System.Net.Http 4.1.1.0 is in the output directory but no binding redirects generated without those properties set.

Repro:
Clone https://github.com/onovotny/azure-mobile-apps-net-client
Checkout 54234e1b4d26737c8d7ae21a29cba694fc2bff0f

Build Microsoft.WindowsAzure.Mobile.Test.Unit and try to run it in the Test Explorer window.

Some of the tests fail with 

Message: Test method Microsoft.WindowsAzure.MobileServices.Test.Unit.Table.Sync.Queue.Actions.PushActionTests.ExecuteAsync_DoesNotSaveTheResult_IfPresentButResultDoesNotHaveId threw exception: 
System.MissingMethodException: Method not found: 'Void Microsoft.WindowsAzure.MobileServices.MobileServiceInvalidOperationException..ctor(System.String, System.Net.Http.HttpRequestMessage, System.Net.Http.HttpResponseMessage)'.

/cc @karelz @ericstj

**Comment by codito** [comment]
Fixed with 758679f5b3f55d9a8b1cc2c696a2fe48e24ef6b1? @Faizan2304 please confirm and close.

**Comment by Faizan2304** [comment]
Yes the issue has been fixed.
